cmd/contour/serve.go

```diff
@@ -448,7 +452,7 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 		log.Info("starting informers")
 		defer log.Println("stopped informers")
 
-		if err := clients.StartInformers(stop); err != nil {
+		if err := clients.StartInformers(serveContext); err != nil {
```

Comment:
If you switch to `g.AddContext(...)` for this task, then the workgroup will pass in a context, and will manage cancellation of that context at the appropriate time.